### DISCUSSION TITLE
NodeJS Load Document Projection

### DISCUSSION THREAD

**Discussion by CyberCyclone** [ravendb/ravendb#13079] [Q&A] [answered]
Hi,

Can someone please let me know how I can do a projection on a `load` statement in NodeJS? I can't seem to find it anywhere in the documentation. If it's not possible, is it because the `load` is designed to be altered and then saved which means the data needs to be there?

```const dbResult = await session.load<StakePool>(`stakePool/${poolHash}`)```

Thanks!
Casey.

**Comment by ayende** [comment, marked as answer]
No, that is not possible. It is something that goes against the concept of what a `load` is.
When you `load` a document, you get _that_ document, and any mutations on that document will be persisted when you call `saveChanges`.

A projection is something else entirely, it is some part of the document (or related ones) and changes to it aren't persisted.

If you want to project a document by id, you need to use `query` for that.

Note that internally, a query by id `where id() == 'stakePool/00000000000000000004d54997799ce846b5526053241e3701ed2296eea5762b` is an operation that will not use an index. Instead, the database will run the same code path for `load` (most efficient one) and then project the data out.
